frontend/public/components/storage-class-form.tsx

```diff
@@ -55,7 +57,105 @@ export class StorageClassForm_ extends React.Component<
     this.previousName = '';
   }
 
-  storageTypes = Object.freeze({
+  defaultProvisionerObj = {
+    title: '',
+    provisioner: '',
+    parameters: {},
+  };
+
+  CSIStorageTypes = {
+    'rbd.csi.ceph.com': {
+      title: 'Ceph RBD',
+      provisioner: 'rbd.csi.ceph.com',
+      documentationLink: 'https://rook.github.io/docs/rook/master/ceph-block.html',
+      parameters: {
+        clusterID: {
+          name: 'Cluster ID',
+          hintText: 'Cluster ID is the namespace where Ceph is deployed.',
+          required: true,
+        },
+        pool: {
+          name: 'Pool',
+          hintText: 'Ceph pool into which the RBD image shall be created',
+          required: true,
+        },
+        imageFormat: {
+          name: 'Image Format',
+          hintText: 'Ceph RBD image format.',
+          required: true,
+        },
+        imageFeatures: {
+          name: 'Image Features',
+          hintText: 'Ceph RBD image features.',
```

Comment:
currently, only layering is supported for image feature https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/ceph/csi/rbd/storageclass.yaml#L28